scripts/test-contracts.ts

```
@@ -1,32 +1,95 @@
-    if (!response.ok) {
-      throw new Error(`HTTP error! status: ${response.status}`);
+async function fetchJson(path: string): Promise<unknown> {
+  const controller = new AbortController();
+  const timeout = setTimeout(() => controller.abort(), 10_000);
+  try {
+    const response = await fetch(`${SITE_URL}${path}`, { headers, signal: controller.signal });
+    if (!response.ok) throw new Error(`${path} HTTP ${response.status}`);
+    return response.json();
+  } catch (err) {
+    if (err instanceof Error && err.name === 'AbortError') {
+      throw new Error(`Timeout after 10s — ${path}`);
```

ESLint

[error] There is no `cause` attached to the symptom error being thrown.

(preserve-caught-error)